website/static/js/newAndNoteworthyPlugin.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * New and Noteworthy Projects
+ */
+
+var $ = require('jquery');
+var m = require('mithril');
+var $osf = require('js/osfHelpers');
+
+// CSS
+require('css/quick-project-search-plugin.css');
+
+// XHR config for apiserver connection
+var xhrconfig = function(xhr) {
+    xhr.withCredentials = true;
+};
+
+
+var newAndNoteworthy = {
+    controller: function() {
+        var self = this;
+        self.newAndNoteworthyNodes = m.prop([]);
+        self.popularNodes = m.prop([]);
+        self.contributorsMapping = {};
+
+        // Load new and noteworthy nodes
+        var newAndNoteworthyUrl = $osf.apiV2Url('nodes/' + window.contextVars.newAndNoteworthy + '/node_links/', {});
+        var newAndNoteworthyPromise = m.request({method: 'GET', url: newAndNoteworthyUrl, config: xhrconfig});
+        newAndNoteworthyPromise.then(function(result){
+            var numPopular = Math.min(result.data.length - 1, 4);
+            for (var l=0; l <= numPopular; l++) {
+                self.newAndNoteworthyNodes().push(result.data[l]);
+                self.fetchContributors(result.data[l]);
+                  }
+            return newAndNoteworthyPromise;
+        });
+
+        // Load popular nodes
+        var popularUrl = $osf.apiV2Url('nodes/' + window.contextVars.popular + '/node_links/', {});
+        var popularPromise = m.request({method: 'GET', url: popularUrl, config: xhrconfig});
+        popularPromise.then(function(result){
+            var numPopular = Math.min(result.data.length - 1, 4);
+            for (var l=0; l <= numPopular; l++) {
+                self.popularNodes().push(result.data[l]);
+                self.fetchContributors(result.data[l]);
+                  }
+            return popularPromise;
+        });
+
+        // Additional API call to fetch node link contributors
+        self.fetchContributors = function(nodeLink) {
+            var url = nodeLink.embeds.target_node.data.relationships.contributors.links.related.href;
+            var promise = m.request({method: 'GET', url : url, config: xhrconfig});
+            promise.then(function(result){
+                var contribNames = [];
+                result.data.forEach(function (contrib){
+                    contribNames.push(contrib.embeds.users.data.attributes.family_name);
+                });
+                var numContrib = result.links.meta.total;
+                var nodeId = nodeLink.id;
+                self.contributorsMapping[nodeId] = [contribNames, numContrib];
+            });
+        };
+
+        // Gets contrib family name for display
+        self.getFamilyName = function(i, node) {
+            return self.contributorsMapping[node.id][0][i];
+        };
+
+        // Returns name if one contrib, or adds et al if > 1
+        self.contribNameFormat = function(node, number) {
```

Comment:
`contribNameFormat` takes a number argument, so that part is easy enough. Options for abstract out the `getFamilyName` function would be: a) funky subclassing so that `self.getFamilyName` inside of `contribNameFormat` resolves differently depending on the calling context, or b) making getFamilyName a third parameter to `contribNameFormat`.

@pattisdr Does that get you pointed in the right direction?